release-22.2: backupccl: upgrade restoreTPCCIncLatest/nodes=10 backup fixture

msbutler opened the pull request:
Backport 1/1 commits from #90669.

/cc @cockroachdb/release

---

restoreTPCCIncLatest/nodes=10 has been consistently failing because the backup fixture is more than one major version behind the cluster running the restore. This patch updates the backup fixture used by this roachtest.

Fixes #88090

Release note: None

Release justification: test only change, fixes a failing roachtest

msbutler committed:
backupccl: deflake roachtest.restore2TB/nodes=10/with-pause

This test pauses a restore several times, but would fail if it attempted to
pause the job after the job already succeeded. This patch prevents the test
from failing if this occurs.

Informs #90023

Release note: None

msbutler committed:
backupccl: upgrade restoreTPCCIncLatest/nodes=10 backup fixture

restoreTPCCIncLatest/nodes=10 has been consistently failing because the backup
fixture is more than one major version behind the cluster running the
restore. This patch updates the backup fixture used by this roachtest. Note
that this fixture was created on 22.2.0-alpha.4, which means that on master,
we're technically doing a "cross version" restore. We decided not to re-create
a fixture for master because a backup taken from master will change
significantly over the release cycle. Once a release SHA is decided for 22.2.0,
the fixtures for master will be updated again.

This patch also updates the tpccInc harness to use the new fixture.

Informs #88090 #89380

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/90856)
<!-- Reviewable:end -->

adityamaru commented on pkg/cmd/roachtest/tests/restore.go:
```diff
@@ -612,6 +565,17 @@ func registerRestore(r registry.Registry) {
 						t.L().Printf("pausing RESTORE job")
 						// Pause the job and wait for it to transition to a paused state.
 						_, err = conn.ExecContext(ctx, `PAUSE JOB $1`, jobID)
+						if err != nil {
```
I think we should error out if we're not able to pause atleast once? Otherwise this test is not doing what it says it is, wdyt

msbutler replied:
good idea. Will send the patch to master and then add it to the backports.

msbutler replied:
opened this on master https://github.com/cockroachdb/cockroach/pull/91024

msbutler replied:
added cherry pick to this branch

msbutler committed:
backupccl: ensure restore2TB/nodes=10/with-pause pauses at least once

This patch adds a check that will cause the roachtest to fail if the job was
not paused at least once.

Release note: None

Epic: none
